### PR TITLE
[RFR] fix of issue when new events aren't compared against all expected events

### PR DESCRIPTION
**Pull request by izapolsk**
{{ pytest: --long-running --use-provider vsphere65-nested cfme/tests/cloud_infra_common/test_events.py }}

There is bug found in EventListener found by @miha-plesko. This PR intends to fix it.

### CODE REVIEW

File: cfme/utils/events.py [hunk 1]
```diff
@@ -135,13 +135,11 @@ def __init__(self, appliance):
...
 
-    def set_last_record(self):
-        """ Sets last_processed_id to the latest event."""
-        last_event_stream = self.event_streams.query_string(limit=1,
-                                                            sort_order='desc',
-                                                            sort_by='id')
-        if len(last_event_stream):
-            self._last_processed_id = last_event_stream[0].id
+    def get_max_record_id(self):
+        try:
```
Comment (miha-plesko) [inline cfme/utils/events.py]: I'd prefer without try catch

```python
evts = self.event_streams.query_string(limit=1, sort_order='desc', sort_by='id')
return evts[0].id if evts else None
```

but it's a matter of preference i guess.

File: cfme/utils/events.py [hunk 2]
```diff
@@ -234,27 +236,29 @@ def process_events(self):
...
+    def get_next_portion(self, evt, max_id=None):
         """ Returns list with one or more events matched with expected event.
 
         Returns None if there is no matched events."""
         evt.process_id()
 
-        q = Q('id', '>', self._last_processed_id)  # ensure we get only new events
+        base_q = Q('id', '>', self._last_processed_id)  # ensure we get only new events
+        if max_id:
+            base_q &= Q('id', '=<', max_id)
```
[the remark this answers is not in the document]
Comment (izapolsk) [reply cfme/utils/events.py]: yep, I already noticed that during testing and fixed. Thanks

File: cfme/utils/events.py [hunk 3]
```diff
@@ -234,27 +236,29 @@ def process_events(self):
...
 
-    def get_next_portion(self, evt):
+    def get_next_portion(self, evt, max_id=None):
         """ Returns list with one or more events matched with expected event.
 
         Returns None if there is no matched events."""
         evt.process_id()
 
-        q = Q('id', '>', self._last_processed_id)  # ensure we get only new events
+        base_q = Q('id', '>', self._last_processed_id)  # ensure we get only new events
```
[the remark this answers is not in the document]
Comment (izapolsk) [reply cfme/utils/events.py]: perhaps you're right

**Comment by izapolsk** [comment]
@miha-plesko, your remarks have been fixed. I'm waiting for remaining reviews. Thank you for help !

**Comment by mshriver** [comment]
PRT directive was set wrong, modified and restarted PRT

**Comment by izapolsk** [comment]
did I write py.test ?

**Comment by mshriver** [comment]
PRT passing, but not collecting the test against either version, appears due to PRT image missing supportability.yaml as the test module should collect against the provider (s) specified.

**Pull request merged**